docs/lifecycle-management.md

```diff
@@ -0,0 +1,185 @@
+# Higher level VM Management
+
+## Motivation
+
+So far KubeVirt provides a pod-like interface when managing VMs. This seems to
+be a good building block.
+On the other hand this might not be sufficient for usual virtualization
+management UIs, which do expect to manage the complete life-cycle of a VM,
+including shut down VMs 8which is currently not intended by the core runtime).
+
+
+This document outlines how minimal assumptions can be met and also visits how
+other requirements impact the design.
+
+
+## Requirements
+* Running and non-running VMs are handled through the same API
+* Creating a VM must be separate from launching a VM
+* Support for “flavor” based VMs (templating)
+* Support for expanding sparse VM definitions
+* Support for detailed VM definitions
+* Support hot-plugging
+* Representation of dynamic instance specific data
+* Support for defaulting (to support OS compatibility)
+Considerations
+* Sub-resources for actions
+* Presets to add certain configurations
+
+
+## Definitions
+* **Defaulting** - Setting certain defaults during expansion (i.e. virtio for a
+  bus type, or e1000 for as a nic model)
+* **Expansion** - VM definitions can be provided in a sparse way i.e. just a
+  disk. Once the VM is getting started, the runtime will infer additional
+  requirements for this device and add them to the runtime VM definition.
+  This process of inferring and adding additional devices is called expansion.
+* **Flavor** - Comparable to a template or a specific instance type
+* **Preset** - Presetting certain values in a specified path of the document
+  (i.e. setting .spec.domain.devices.nvram). It is not about setting certain
+  default values for undefined fields (i.e. set virtio for all busses)
```

Comment:
> I would suggest to rethink defaulting...to me it also seems as something which is not really needed - if there is a 1:1 mapping between what you express in spec and the details(e.g. an expansion from distro suse-10 leading to addition of virtio-blk controller) and you want to express the derived items in the spec for customization(like the controller type, constrained by what libosinfo defines), then why do you need anything else than a preset?

In the scope of this document, and considering the presets, we _do_ have the functionality to add certain elements (i.e. a disk or device with the specific attributes, like the type) if a certain label/metadata is set.
But there is (without defaulting) now way to say that a virtio-blk controller should be used by default, if a controller iwth out a type or a device without a controller is added to the spec and needs to be inferred.
To cover this second case I made the suggestion above to make the defaults part of the spec, because then it might be efficient if we combine this with presets.

The flow could be:
1. libosinfo to create distro specific presets containing defaults
2. user creates vm spec, and sets the correct metadata i.e. `distro: suse-10`
3. Upon start the config is expanded into an instance using defaults specified in the defaults section of the spec

@michalskrivanek does this make sense?